src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs

```diff
@@ -272,43 +279,56 @@ public override Task<bool> FixAsync(ICodeCleanUpScope scope, ICodeCleanUpExecuti
 
         private Task<bool> FixHierarchyContentAsync(IVsHierarchyCodeCleanupScope hierarchyContent, ICodeCleanUpExecutionContext context, CancellationToken cancellationToken)
         {
-            // TODO: this one will be implemented later
-            // https://github.com/dotnet/roslyn/issues/30165
             var hierarchy = hierarchyContent.Hierarchy;
             if (hierarchy == null)
             {
-                // solution
-                return Task.FromResult(false);
+                return FixSolutionAsync(_workspace.CurrentSolution, context, cancellationToken);
             }
 
             var itemId = hierarchyContent.ItemId;
             if (itemId == (uint)VSConstants.VSITEMID.Root)
             {
-                // Project
+                var hierarchyToProjectMap = _workspace.Services.GetRequiredService<IHierarchyItemToProjectIdMap>();
+                var projectHierarchyItem = _vsHierarchyItemManager.GetHierarchyItem(hierarchyContent.Hierarchy, itemId);
+                if (!hierarchyToProjectMap.TryGetProjectId(projectHierarchyItem, targetFrameworkMoniker: null, out var projectId))
+                {
+                    return Task.FromResult(false);
+                }
+
+                var project = _workspace.CurrentSolution.GetProject(projectId);
+                if (project is null)
```

Comment:
"==" ?

Reply:
It was the same here, but I fixed it while I was cleaning up other things.